### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by unirho**

### CODE REVIEW

File: addons/udes_warehouse_classification/tests/test_product_template.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        super().setUpClass()
+
+        cls.test_messages_01 = ["Inbound, Report a, Report"]
+        cls.test_messages_02 = ["Report a Report", "Report b Report"]
+        cls.test_messages_03 = ["Report a Report", "Inbound, Report a, Report"]
+
+    # functions to test:  and _get_classification_messages_for_product_picking
+
+    # For a given product find the classification messages for specified report,
+    # 1. report a on it's own - banana
```
Comment (nocturnalastro) [inline addons/udes_warehouse_classification/tests/test_product_template.py:16]: Either put this as the docstring of the class or add it to the docstring of each test :)

File: addons/udes_warehouse_classification/tests/test_product_template.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    # 2. not matching report type - banana
+    # 3. report a and b - honeydew
+    # 4. report a and report a on inbound (i.e. 2 messages for a report) - cherry
+    # 5. no classifications - apple
+
+    def test01_basic_report(self):
+        messages = self.banana.product_tmpl_id.get_classification_messages_for_report(
```
Comment (nocturnalastro) [inline addons/udes_warehouse_classification/tests/test_product_template.py:23]: why `banana.product_tmpl_id.get_classification_messages_for_report` and not just `banana.get_classification_messages_for_report`?

File: addons/udes_warehouse_classification/tests/test_product_template.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    def test05_no_classfications_for_product(self):
+        with self.assertLogs(
+            "odoo.addons.udes_warehouse_classification.models.product_template"
+        ) as cm:
+            messages = self.apple.product_tmpl_id.get_classification_messages_for_report(
+                self.report_b.report_name
+            )
+        self.assertFalse(messages)
+        self.assertIn("Product Test product Apple has no warehouse classifications.", cm.output[0])
```
Comment (nocturnalastro) [inline addons/udes_warehouse_classification/tests/test_product_template.py:59]: What about multiple products? It is api.multi after all

File: addons/udes_warehouse_classification/tests/test_stock_picking.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
+from .common import Base
+
+
+class TestWarehouseClassificationPicking(Base):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.test01_messages = {"productDamson": [{"message": "Inbound Alert"}]}
```
Comment (nocturnalastro) [inline addons/udes_warehouse_classification/tests/test_stock_picking.py:9]: Is this the expected result for each test? If its test specific add it to the test, please :)

File: addons/udes_warehouse_classification/tests/test_product_template.py (hunk 2 above)
Comment (unirho) [reply addons/udes_warehouse_classification/tests/test_product_template.py]: made a proxy method from product.product

File: addons/udes_warehouse_classification/tests/test_product_template.py (hunk 3 above)
Comment (unirho) [reply addons/udes_warehouse_classification/tests/test_product_template.py:62]: fixed

**Review by nocturnalastro — APPROVED** [review]

File: addons/udes_warehouse_classification/tests/test_product_template.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class TestWarehouseClassificationPicking(Base):
+    """
+        # For a given product find the classification messages for specified report,
+        # 1. report a on it's own - banana
+        # 2. not matching report type - banana
+        # 3. report a and b - honeydew
+        # 4. report a and report a on inbound (i.e. 2 messages for a report) - cherry
+        # 5. no classifications - apple
```
Comment (joe-gb) [inline addons/udes_warehouse_classification/tests/test_product_template.py]: 5 tests in docstring but 6 below! did one get added?

**Review by joe-gb — APPROVED** [review]

File: addons/udes_warehouse_classification/tests/test_product_template.py (hunk 5 above)
Comment (unirho) [reply addons/udes_warehouse_classification/tests/test_product_template.py]: fixed

File: addons/udes_warehouse_classification/tests/common.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+    @classmethod
+    def create_classification(
+        cls, name, alert_message=False, report_message=False, picking_types=None, report_types=None
+    ):
+
+        Classification = cls.env["product.warehouse.classification"]
+        classification_info = {
+            "name": name,
+            "alert_message": alert_message or False,
```
Comment (Jason-Scott-23) [inline addons/udes_warehouse_classification/tests/common.py]: Remove `or False` as `alert_message` default value is `False`.

**Review by Jason-Scott-23 — APPROVED** [review]

**Pull request merged**